src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,434 @@
+        {
+            Application.Current.ThemeMode = themeMode;
+            return true;
+        }
+        return false;
+    }
+
+    internal static void SyncDeferredThemeModeAndResources()
+    {
+        if(Application.Current == null) return;
+
+        ThemeMode themeMode = Application.Current.ThemeMode;
+
+        bool resyncThemeMode = false;
+        int index = FindLastFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+
+        if(index > 0)
+        {
+            ResourceDictionaryContainsFluentDictionary(Application.Current.Resources, out ThemeMode _themeMode);            
+            resyncThemeMode = true;
+            themeMode = _themeMode;
+        }
+        else
+        {
+            if(themeMode != ThemeMode.None && index != 0)
+            {
+                resyncThemeMode = true;
+            }
+
+            if(themeMode == ThemeMode.None && index == 0)
+            {
+                resyncThemeMode = true;
+            }
+        }
+
+        if(resyncThemeMode)
+        {
+            Application.Current.ThemeMode = themeMode;
+        }
+    }
+
+    internal static void ApplyStyleOnWindow(Window window)
+    {
+        if(!IsFluentThemeEnabled && window.ThemeMode == ThemeMode.None) return;
+
+        bool useLightColors = GetUseLightColors(Application.Current.ThemeMode);
+        if(window.ThemeMode != ThemeMode.None)
+        {
+            useLightColors = GetUseLightColors(window.ThemeMode);
+        }
+        ApplyStyleOnWindow(window, useLightColors);
+    }
+
+    internal static bool IsValidThemeMode(ThemeMode themeMode)
+    {
+        return themeMode == ThemeMode.None || themeMode == ThemeMode.Light || themeMode == ThemeMode.Dark || themeMode == ThemeMode.System;
+    }
+
+    internal static Uri GetThemeResource(ThemeMode themeMode)
+    {
+        bool useLightColors = GetUseLightColors(themeMode);
+        return GetFluentThemeResourceUri(useLightColors);
+    }
+
+    #endregion
+
+
+    #region Private Methods
+
+    private static void RemoveFluentFromApplication()
+    {
+        if(Application.Current == null) return;
+
+        List<int> indices = FindAllFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+
+        foreach(int index in indices)
+        {
+            Application.Current.Resources.MergedDictionaries.RemoveAt(index);
+        }
+
+        foreach(Window window in Application.Current.Windows)
+        {
+            if(!FluentEnabledWindows.HasItem(window))
+            {
+                RemoveStyleFromWindow(window);
+            }
+        }
+    }
+
+    private static void RemoveFluentFromWindow(Window window)
+    {
+        if(window == null || window.IsDisposed) return;
+
+        List<int> indices = FindAllFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+
+        foreach(int index in indices)
+        {
+            Application.Current.Resources.MergedDictionaries.RemoveAt(index);
```

Comment:
Can I know why the `RemoveFluentFromWindow` will do something like `RemoveFluentFromApplication`

Reply:
Fixing this is next commit.